### PR TITLE
Hayley add res data

### PR DESCRIPTION
**Pull request by hcorson-dosch-usgs**
Making a draft PR b/c @limnoliver and @aappling-usgs it would be great to get your feedback before I finish adding in some minimal metadata (many of the files only have placeholder metadata for now)

I've pushed everything [to SB](https://www.sciencebase.gov/catalog/item/5f6a26af82ce38aaa2449100), so that you can view it there. There were a few things I was hoping to check with you about:
* Right now I'm rendering the xml for `00_parent`, `01_spatial` and `02_observations` with their associated in_text as well as  `river_metadata` (derived from `network_vertices_sf`) and `reservoir_metadata` (derived from `reservoir_polygons`). I wasn't sure if this was the right way to deal w/ multiple `sf` objects, and I'm thinking it's not, b/c on SB only the extent of `reservoir_polygons` is shown as the footprint. Is there a better way to define two different extents? Or should I only use the larger extent (`river_metadata`)?
* Before I sketch in minimal metadata for Alison's added reservoir modeling files (for Pepacton and Cannonsville) and for Sam's Lordville subsets of the data already included in the first version of this release, does the way I've structured the targets seem correct? Or should I revise any of the target locations or names?

### CODE REVIEW

**Comment by limnoliver** [comment]
> Right now I'm rendering the xml for 00_parent, 01_spatial and 02_observations with their associated in_text as well as river_metadata (derived from network_vertices_sf) and reservoir_metadata (derived from reservoir_polygons). I wasn't sure if this was the right way to deal w/ multiple sf objects, and I'm thinking it's not, b/c on SB only the extent of reservoir_polygons is shown as the footprint. Is there a better way to define two different extents? Or should I only use the larger extent (river_metadata)?

I would like this to show the larger extent (so river_metadata). Jordan [has shown an extent with polygons](https://www.sciencebase.gov/catalog/item/5ebe564782ce476925e44b26), so maybe that is our best bet here -- to show the larger basin extent, with river lines + two lake polygons? Here is the [code for that release as an example](https://github.com/USGS-R/pgmtl-data-release). I would also say this is low priority to figure out, and would become more important if this turns into a published data release to support modeling work by Xiaowei. 

As far as organization goes, I think where you put the targets seems right! I added just one comment about reservoir file naming conventions below.

File: 2_observations.yml [hunk 1]
```diff
@@ -28,10 +36,31 @@ targets:
...
   out_data/reservoir_releases_lordville.csv:
     command: filter_reservoirs(out_file = target_name, in_dat = 'out_data/reservoir_releases.csv', keep = I(c('Pepacton', 'Cannonsville')))
     
   # get GRAND IDs of reservoirs above Lordville
   res_lordville:
-    command: get_grands(in_file = 'out_data/reservoir_releases_lordville.csv')
\ No newline at end of file
+    command: get_grands(in_file = 'out_data/reservoir_releases_lordville.csv')
+
+  out_data/res_io_obs.feather:
```
Comment (limnoliver) [inline 2_observations.yml]: Are these ins/outs a combo of model predictions and true observations? Thinking about where these should go if they are a mix. I think here is okay even if they contain some model predictions.

File: 2_observations.yml [hunk 2]
```diff
@@ -28,10 +36,31 @@ targets:
     
   out_data/reservoir_releases.csv:
     command: file.copy(from = '../delaware-model-prep/2_observations/out/reservoir_releases.csv', to = target_name)
-  
+
   out_data/reservoir_releases_lordville.csv:
```
Comment (limnoliver) [inline 2_observations.yml:41]: maybe keep consistent naming conventions when it is a reservoir file? I think now we're going between "reservoir" and "res"

File: 2_observations.yml (hunk 1 above)
Comment (limnoliver) [reply 2_observations.yml]: Nevermind -- I see the SNTEMP predictions below in predictions. Nice!

**Comment by hcorson-dosch-usgs** [comment]
Ok - I've updated everything [on SB](https://www.sciencebase.gov/catalog/item/5f6a26af82ce38aaa2449100)

@limnoliver -- I think I address all your comments re: spatial metadata and naming conventions

@aappling-usgs -- I added in targets for the realsat surface area data as well as the interpolated daily diversions, and also added minimal metadata for all of the files on SB.

File: 1_spatial.yml [hunk 3]
```diff
@@ -58,4 +61,23 @@ targets:
...
+      in_ind = "../lake-temperature-model-prep/1_crosswalk_fetch/out/canonical_lakes_sf.rds.ind",
+      in_repo = I('../lake-temperature-model-prep/'),
+      site_ids = reservoir_modeling_site_ids)
+      
+  reservoir_metadata:
+    command: extract_feature(reservoir_polygons)
+    
+  out_data/01_reservoir_polygons.zip:
+    command: sf_to_zip(zip_filename = target_name, 
+      sf_object = reservoir_polygons, layer_name = I('study_reservoirs'))
```
Comment (limnoliver) [inline 1_spatial.yml:83]: Currently there are no data on the [spatial page](https://www.sciencebase.gov/catalog/item/5f6a285d82ce38aaa244912e). Is that expected?

**Review by limnoliver — APPROVED** [review]

File: 1_spatial.yml (hunk 3 above)
Comment (hcorson-dosch-usgs) [reply 1_spatial.yml:83]: Ugh, no. Science Base was being buggy last night - I'll push those again now

Comment (hcorson-dosch-usgs) [reply 1_spatial.yml:83]: Ok the 2 zip files should be there now

**Review by aappling-usgs — APPROVED** [review]

**Comment by aappling-usgs** [comment]
I'm ready for you to merge if you are.

**Pull request merged**